### PR TITLE
Use int64_t as offset to ext4_fseek.

### LINKED ISSUE #25: ext4_fseek offset is unsigned
Generally `fseek` takes a signed offset; this allows seeking backward when using `SEEK_CUR` and `SEEK_END`. `ext4_fseek` uses a `uint64_t` offset, so it's not possible to seek backwards unless you do the math manually with `ext4_ftell` and then use `SEEK_SET`.

### PR DESCRIPTION
**Pull request by enetor**
This change makes it possible to fseek backwards in fseek. It fixes #25 

Please let me know if you would like to have a specific test for the offsets and different origins.

Tested:
  make test_all

### CODE REVIEW

File: src/ext4.c [hunk 1]
```diff
@@ -2016,23 +2016,25 @@ int ext4_fwrite(ext4_file *file, const void *buf, size_t size, size_t *wcnt)
...
-		if (offset > file->fsize)
+		if (offset < 0 || (uint64_t)offset > file->fsize)
 			return EINVAL;
 
 		file->fpos = offset;
 		return EOK;
 	case SEEK_CUR:
-		if ((offset + file->fpos) > file->fsize)
+		if ((offset < 0 && (uint64_t)(-offset) > file->fpos) ||
+		    (offset > 0 &&
```
Comment (gkostka) [inline src/ext4.c]: This line is redundant IMO.

File: src/ext4.c [hunk 2]
```diff
@@ -2016,23 +2016,25 @@ int ext4_fwrite(ext4_file *file, const void *buf, size_t size, size_t *wcnt)
...
+		     (uint64_t)offset > (file->fsize - file->fpos)))
 			return EINVAL;
 
 		file->fpos += offset;
```
Comment (gkostka) [inline src/ext4.c]: It is equivalent to ```file->fpos = file->fpos + offset;```. I think we need to check if there is not implicit conversion ```offset``` variable to uint64_t

**Comment by gkostka** [comment]
LGTM but I posted some comments:). If you have some time you could add few tests to test suite. Just to be sure that everething is OK with new fseek implementation. Thanks!

File: src/ext4.c (hunk 1 above)
Comment (enetor) [reply src/ext4.c]: Example:

offset = -10
file->fpos = 12
file->fsize = 13

This offset is legit. If we omit line 2030:

if ((offset < 0 && (uint64_t)(-offset) > file->fpos)  ||   // evaluates to false
    (uint64_t)offset > (file->fsize - file->fpos))              // evaluates to true
Returns EINVAL

I will add a test case for this.

File: src/ext4.c (hunk 2 above)
Comment (enetor) [reply src/ext4.c]: That's right. Let me work on the test cases first. Then I will fix this.

File: src/ext4.c (hunk 1 above)
Comment (gkostka) [reply src/ext4.c]: Ups. Sorry, i'm on holiday and I don't think think clearly:)

File: src/ext4.c (hunk 2 above)
Comment (gkostka) [reply src/ext4.c]: It sholud work without casting if we assume complement two encoding of signed values.

Comment (enetor) [reply src/ext4.c]: Hmm, in Linux ext4 driver file->fpos is signed and makes everything much easier. I think that might be easier and more error-prone to also make fpos signed here. Mixing unsigned and signed is almost always a bad idea...

Comment (gkostka) [reply src/ext4.c]: Changing file->fpos (and maybe file->fsize) to int64 seems to be a good idea. IMO we should do it as a separate PR.

**Pull request merged**